src/libraries/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -479,17 +465,17 @@ private void CreateFromBinaryForm(byte[] binaryForm, int offset)
                 (((long)binaryForm[offset + 6]) << 8) +
                 (((long)binaryForm[offset + 7])));
 
-            SubAuthorities = new int[binaryForm[offset + 1]];
+            int subAuthoritiesLength = binaryForm[offset + 1];
```

Comment:
It is possible this is 0?  If yes, then the change made in CreateFromParts is going to throw an exception because the span will be empty, whereas previously it would have accepted a non-null but empty array.

Reply:
Unless something already would have caught it, it's entirely possible... because this is interpreting data from a public ctor (`SecurityIdentifier..ctor(byte[], int)`).

Reply:
It can be 0 and valid. I've added a max length check and test.